extras/i2c/i2c.h

```diff
@@ -27,16 +27,39 @@
 
 #include <stdint.h>
 #include <stdbool.h>
+#include <errno.h>
+#include <FreeRTOS.h>
+#include <task.h>
 
 #ifdef	__cplusplus
 extern "C" {
 #endif
 
+
+#define I2C_FREQUENCY_400K true  // for test WIP
+
+/*
+ *  Some bit can be transmit slower.
+ *  Selected frequency fix the speed of a bit transmission
+ *  I2C lib take the maximum frequency defined
+ */
+
+#ifdef I2C_FREQUENCY_500K
+#define I2C_CUSTOM_DELAY_160MHZ   6
+#define I2C_CUSTOM_DELAY_80MHZ    1   //Sry, maximum is 320kHz at 80MHz
```

Comment:
Something that could be done to squeeze the last bit of performance and go up to 400 kHz is avoiding `sdk_system_get_cpu_freq` call in the delay function. Instead, the state of cpu frequency can be cached into a `static uint8_t` variable at the beginning of the transaction. This way, there will be no actual `call0` performed at every delay call, only `l32r`, `l32i` and a comparison. The downside is that CPU frequency should not change during i2c transactions — a reasonable requirement IMO. 

I'm not saying you should implement this in this PR — you already did a lot of work. Just for future reference. I'm out of esp modules currently but I'll try to do it myself once I receive a new batch.